enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+
+A new controller in the cluster-etcd-operator would then reconcile this CRD to save backups locally per the specified configuration.
+
+### API Extensions
+
+A new CRD `EtcdBackup` will be introduced to the API group `config.openshift.io` with the version `v1alpha1`. The CRD would be feature-gated until the prerequisite e2e test has an acceptable pass rate. 
+
+The spec and status structures will be as listed below:
+
+
+```Go
+// EtcdBackupSpec represents the configuration of the EtcdBackup CRD.
+type EtcdBackupSpec struct {
+
+    // Reason defines the reason for the most recent backup.
+    // Setting this to a value different from the most recent reason will trigger a one-time backup
+    // The default "" means no backup is triggered
+    // +kubebuilder:default:=""
+    // +optional
+    Reason string `json:"reason"`
+    
+    // Schedule defines the recurring backup schedule in Cron format
+      // every 2 hours: 0 */2 * * *
+      // every day at 3am: 0 3 * * *
+    // Setting to an empty string "" means disabling scheduled backups
+    // Default: ""
+    // TODO: Define how we do validation for the format (e.g validation webhooks)
+    // and the limits on the frequency to disallow unrealistic schedules (e.g */2 * * * * every 2 mins)
+    // TODO: What is the behavior if the backup doesn't complete in the specified interval
+        // e.g: every 1hr but the backup takes 2hrs to complete
+        // Wait for the current one to complete?
+        // And do we generate an alert even if MaxDurationToCompleteBackup is not set, since the scheduled interval has passed?
+    Schedule string `json:"schedule"`
+    
+    // MaxDurationToCompleteBackup sets the max duration after which if
+    // an initiated backup has not successfully completed a critical alert will be generated.
+    // The format is as accepted by time.ParseDuration(), e.g "1h10m"
+    // See https://pkg.go.dev/time#ParseDuration
+    // Setting to "0" disables the alert.
+    // Default: "1h"
+    // TODO: Validation on min and max durations.
+        // The max should be smaller than the scheduled interval
+        // for the next backup else the alert will never be generated
+    MaxDurationToCompleteBackup string `json: "maxDurationWithoutBackup"`
+    
+    // RetentionCount defines the maximum number of backups to retain.
+    // If the number of successful backups matches retentionCount 
+    // the oldest backup will be removed before a new backup is initiated.
+    // The count here is for the total number of backups
+    // TODO: Validation on min and max counts.
+    // TODO: Retention could also be based on the total size, or time period
+        // e.g retain 500mb of backups, or discard backups older than a month
+        // If we want to extend this API in the future we should make retention
+        // a type so we can add more optional fields later
+    RetentionCount int `json: "retentionCount"`
```

Comment:
if we already have good ideas on other types of retention, maybe add a struct for this? I know it's TP MVP, but let's not introduce something we have to completely change in v1alpha2

Reply:
>if we already have good ideas on other types of retention, maybe add a struct for this? I know it's TP MVP, but let's not introduce something we have to completely change in v1alpha2

Agreed, you want to get close on your first attempt and this looks ripe for discriminated  union.